src/components/spreadsheet/columns-config.tsx

```diff
@@ -74,14 +77,26 @@ export const ColumnsConfig: FunctionComponent<ColumnsConfigProps> = ({ tabIndex,
     }, [tableDefinition?.columns, handleCloseColumnsSettingDialog]);
 
     const handleSaveSelectedColumnNames = useCallback(() => {
-        dispatch(
-            updateTableDefinition({
-                ...tableDefinition,
-                columns: localColumns,
+        reorderSpreadsheetColumns(
+            tableDefinition.uuid,
+            localColumns.map((col) => col.uuid)
+        )
+            .then(() => {
+                dispatch(
+                    updateTableDefinition({
+                        ...tableDefinition,
+                        columns: localColumns,
+                    })
+                );
             })
```

Comment:
When we are here (after validate) we may have changed a combinaison of lock, visible, or columns reorder. It's a pitie we always sent a back-end request to update the database even if we havent change the columns order. Could we avoid this ?